### DISCUSSION TITLE
Playlists not visible after creation

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1079] [Bug Reports]
rAudio-1: 20221014 - Using the new Shared Data NFS setup on 3RPis

I have been exploring Playlists again and am having some problems. I created 2 playlists, both of which were initially visible in the Playlists screen and via the Playlist icon in the Library on Machine 1.  Switching to Machine2 the Playlist screens (Playlist and Library) showed no Playlists available.  Switching back to Machine1 no playlists were apparently available there either.  Switching to SSH I can see:

Machine1:
RAOffice:/srv/http/data  #  ls -lrt
total 48
lrwxrwxrwx 1 http   http      8 Sep  7 07:22 shm -> /dev/shm
drwxr-xr-x 4 http   http   4096 Sep  7 07:46 camilladsp
drwxr-xr-x 2 http   http   4096 Sep  7 07:46 tmp
drwxr-xr-x 2 http   http   4096 Sep  7 07:46 lyrics
drwxr-xr-x 2 http   http   4096 Sep  7 07:46 bookmarks
drwxr-xr-x 2 http   http   4096 Sep  7 07:46 audiocd
drwxr-xr-x 3 http   http   4096 Oct 10 10:00 backup
drwxr-xr-x 2 mpd    audio  4096 Oct 10 19:56 mpd
drwxr-xr-x 2 http   http   4096 Oct 14 08:25 addons
-rwxrwxrwx 1 nobody nobody   42 Oct 14 08:30 sharedip
drwxr-xr-x 3 http   http   4096 Oct 14 08:33 webradio
drwxr-xr-x 2 http   http   4096 Oct 14 14:22 system
drwxr-xr-x 2 mpd    audio  4096 Oct 21 10:12 playlists

RAOffice:/srv/http/data  #  ls -la playlists
total 48
drwxr-xr-x  2 mpd  audio  4096 Oct 21 10:12  .
drwxrwxrwx 13 http http   4096 Oct 14 08:29  ..
-rw-r--r--  1 mpd  mpd      63 Oct 21 10:08 'Pirate Radio.m3u'
-rw-r--r--  1 mpd  mpd   36188 Oct 21 10:12 'Random Christmas Tracks.m3u'

and on Machine2:
rAudio-1:/srv/http/data  #  ls -lrt
total 20
drwxr-xr-x 4 http http 4096 May 26 12:14 camilladsp
lrwxrwxrwx 1 http http    8 May 26 12:15 shm -> /dev/shm
drwxr-xr-x 2 http http 4096 Jun 22 10:12 tmp
lrwxrwxrwx 1 http http   25 Oct 10 15:47 audiocd -> /mnt/MPD/NAS/data/audiocd
lrwxrwxrwx 1 http http   27 Oct 10 15:47 bookmarks -> /mnt/MPD/NAS/data/bookmarks
lrwxrwxrwx 1 http http   24 Oct 10 15:47 lyrics -> /mnt/MPD/NAS/data/lyrics
lrwxrwxrwx 1 http http   21 Oct 10 15:47 mpd -> /mnt/MPD/NAS/data/mpd
lrwxrwxrwx 1 http http   27 Oct 10 15:47 playlists -> /mnt/MPD/NAS/data/playlists
lrwxrwxrwx 1 http http   26 Oct 10 15:47 webradio -> /mnt/MPD/NAS/data/webradio
drwxr-xr-x 2 http http 4096 Oct 10 15:47 system
drwxr-xr-x 2 http http 4096 Oct 14 08:27 addons
drwxr-xr-x 7 http http 4096 Oct 14 08:33 backup

rAudio-1:/srv/http/data  #  ls -la playlists
lrwxrwxrwx 1 http http 27 Oct 10 15:47 playlists -> /mnt/MPD/NAS/data/playlists
rAudio-1:/srv/http/data  #

From this it appears that Playlists are owned by owner/group 'mpd' where everything else on the share data areas is owned by owner/group 'http'.  Could this be where the problem lies ?

Regards.

Chris

**Comment by ceejayemm** [comment]
After updating to 20221117 this problem still exists.  I can create a Playlist and save it, it is saved to '/srv/http/data/playlists' as noted above. For a little while the new playlist can be seen in Library / Playlists (along with some others created some while ago).  BUT after leaving the Library and doing 'other things' (can't be sure what constitutes 'other things' and I am not sure it even matters) and then returning to Library / Playlists no playlists are visible.

Created playlist:

![playlist1](https://user-images.githubusercontent.com/25418025/203126764-4860273a-85be-4614-b1df-8248e563d66e.png)

Moved to Library to confirm Playlist available, there were 4 now 5:

![playlist2](https://user-images.githubusercontent.com/25418025/203126912-4b54a3e6-d7bd-422f-b8ab-4548db96e42a.png)

Closed the screen grabber, tried to open Playlists is enough to 'lose' all the playlists (they are still there in '/srv/http/data/playlists' but can't be 'seen' in the web app):

rAudio-1:/srv/http/data/playlists  #  ls -lrt
total 148
-rwxr-xr-x 1 http http    63 Oct 21 10:08 'Pirate Radio.m3u'
-rwxr-xr-x 1 http http 36188 Oct 21 10:12 'Random Christmas Tracks.m3u'
-rw-r--r-- 1 mpd  mpd  36188 Nov 21 17:28 'Christmas - Random Tracks.m3u'
-rw-r--r-- 1 mpd  mpd  36188 Nov 21 17:30  Christmas.m3u
-rw-r--r-- 1 mpd  mpd  36188 Nov 21 17:54  Test1.m3u


![playlist3](https://user-images.githubusercontent.com/25418025/203127440-8991e97d-dd0a-4dd1-b5c1-561f1e6e2d4c.png)

Chris

**Comment by rern** [comment]
Hi Chris,

Please try the UPDATE branch.

**Reply by ceejayemm** [reply]
@rern 
Yes, the Update branch seems to solve the problem with being able to access saved playlists.  There is however a minor problem with moving to the Playlist page on the bottom bar. I know there are other ways of doing this but I prefer to use the bottom bars

If I create a new playlist, save it and then go the Library / Playlists page I can see the new Playlist along with all the others I had previously saved.  If I select a Playlist here and select Replace, the screen tells me Playlist Replaced / Done. If I then try to select the Playlist page from the bottom bar  I cannot do this. I have to select the Play page from the bottom bar first and then I can select the Playlist page from the bottom bar. I can then see all the tracks saved in the current playlist as I would expect

Thanks

Chris

**Reply by ceejayemm** [reply]
@rern 
With SharedData (from rAudio-1 server) enabled it is only possible to delete Playlists from the 'server' machine. Is this by design ?

Chris

**Reply by ottl05** [reply]
> ottom

How could I change to the update branch?

**Comment by ottl05** [comment]
I´ve got also a problem with the playlists. in the library I
 see that there are playlists, but I can´t select them, 
There´s only an empty page.

It persist in version rAudio 20221123
